### PR TITLE
crypto: fix Bip44 derivation path

### PR DESCRIPTION
**Pull request by araskachoi**
Closes: #XXX

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: types/config.go [hunk 1]
```diff
@@ -2,6 +2,7 @@ package types
 
 import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	eth "github.com/ethereum/go-ethereum/accounts"
```
Comment (fedekunze) [inline types/config.go]:
```suggestion
	ethaccounts "github.com/ethereum/go-ethereum/accounts"
```

File: types/config.go [hunk 2]
```diff
@@ -23,9 +24,11 @@ const (
 
 	// Bip44CoinType satisfies EIP84. See https://github.com/ethereum/EIPs/issues/84 for more info.
 	Bip44CoinType = 60
+)
 
+var (
 	// BIP44HDPath is the BIP44 HD path used on Ethereum.
-	BIP44HDPath = "44'/60'/0'/0/0"
+	BIP44HDPath = eth.DefaultRootDerivationPath.String()
```
Comment (fedekunze) [inline types/config.go]:
```suggestion
	BIP44HDPath = ethaccounts.DefaultRootDerivationPath.String()
```

File: crypto/hd/algorithm_test.go [hunk 3]
```diff
@@ -122,3 +125,43 @@ func TestKeyring(t *testing.T) {
...
+		log.Fatal(err)
+	}
+
+	// Equality of Address BIP44 ian
+	require.Equal(t, account.Address.Hex(), "0xA588C66983a81e800Db4dF74564F09f91c026351")
+	// Equality of Ethermint implementation
+	fmt.Println(privkey.PubKey().Address())
+	require.Equal(t, "0x"+hex.EncodeToString(privkey.PubKey().Address().Bytes()), strings.ToLower("0xA588C66983a81e800Db4dF74564F09f91c026351"))
+	// // Equality of Eth and Ethermint implementation
+	require.Equal(t, "0x"+hex.EncodeToString(privkey.PubKey().Address()), strings.ToLower(account.Address.Hex()))
```
Comment (fedekunze) [inline crypto/hd/algorithm_test.go]: I'd use BytesToAddress instead of `"0x"+hex.EncodeToString`

File: crypto/hd/algorithm_test.go [hunk 4]
```diff
@@ -122,3 +125,43 @@ func TestKeyring(t *testing.T) {
...
+	// need to prepend "m/" because the below method provided by the sdk does not add the proper prepending
+	hdPath := "m/" + params.String()
+	require.Equal(t, "m/44'/60'/0'/0/0", hdPath)
+}
+
+func TestDerivation(t *testing.T) {
+
+	mnemonic := "picnic rent average infant boat squirrel federal assault mercy purity very motor fossil wheel verify upset box fresh horse vivid copy predict square regret"
+
+	bz, err := DeriveSecp256k1(mnemonic, keys.DefaultBIP39Passphrase, "m/44'/60'/0'/0/0")
```
Comment (fedekunze) [inline crypto/hd/algorithm_test.go]: can you add a test case that compares the addresses generated with and without the `m/` prefix to be not equal?

**Review by fedekunze — CHANGES_REQUESTED** [review]
Thanks!  Minor comments. Also pending a bugfix changelog entry 👍

File: crypto/hd/algorithm_test.go (hunk 4 above)
Comment (araskachoi) [reply crypto/hd/algorithm_test.go]: added the test

**Comment by araskachoi** [comment]
> Thanks! Minor comments. Also pending a bugfix changelog entry 👍

@fedekunze I think its g2g! If you want to give a last look and appove!

**Review by fedekunze — APPROVED** [review]
ACK

**Pull request merged**